discover.py

```diff
@@ -0,0 +1,260 @@
+            return
+
+        rewrite_dnsmasq_conf(app.config['DNSMASQ_CONFIG_FILE'], conf_string)
+        last_dnsmasq_conf_string = conf_string
+        reload_dnsmasq()
+
+    last_dnsmasq_conf_string = ''
+    while True:
+        update_dnsmasq()
+        time.sleep(3)
+
+def validate_hostname(hostname):
+    if not bool(re.match(r"^((\w|-)+\.)+(\w+)$", hostname)):
+        raise ValidationError("invalid hostname: {}".format(hostname))
+
+def validate_ipv4_address(address):
+    if not bool(re.match(r"^(\d{1,3}\.){3}\d{1,3}$", address)):
+        raise ValidationError("invalid ipv4 address: {}".format(address))
+
+def valid_record(interface, hostname, address):
+    try:
+        validate_hostname(hostname)
+        validate_ipv4_address(address)
+    except ValidationError:
+        log.warning("%r skipping record with hostname %r and address %r",
+                    interface, hostname, address)
+        return False
+    else:
+        return True
+
+def get_dns_conf_string(interface):
+    ip = get_ipv4_addr(interface)
+    common_args = [
+        'domain-needed',
+        'bogus-priv',
+        'no-resolv',
+        'no-hosts',
+        'bind-interfaces',
+        'interface={}'.format(interface),
+        'listen-address={}'.format(ip),
+    ]
+
+    dns_servers = [
+        'server=' + dns_server_ip
+        for dns_server_ip in app.config["DNS_SERVERS"]
+    ]
+
+    address_records = [
+        (node['hostname'], node['address'])
+        for node in nodes.get(interface, {}).values()
+        if not node['is_local']
+    ]
+    address_records.sort()
+
+    # add current node as first dns record
+    address_records.insert(0, (app.config['LIQUID_DOMAIN'], ip))
+    address_args = [
+        'address=/{}/{}'.format(hostname, address)
+        for hostname, address in address_records
+        if valid_record(interface, hostname, address)
+    ]
+
+    args = common_args + dns_servers + address_args
+    return "\n".join(args) + "\n"
+
+def normalize_dict_str(data):
+    return {
+        str(key, encoding='latin-1'): str(data[key], encoding='latin-1')
+        for key in data
+    }
+
+def get_ipv4_addr(interface):
+    def valid_addr(a):
+        return 'addr' in a and \
+               a.get('netmask') != HOST_ONLY_NETWORK_MASK
+
+    addrs = netifaces.ifaddresses(interface).get(netifaces.AF_INET)
+    if not addrs:
+        return None
+
+    ips = [
+        a['addr']
+        for a in addrs
+        if valid_addr(a)
+    ]
+
+    if ips:
+        assert len(ips) == 1, "Only one ip per interface is supported."
+        return ips[0]
+    else:
+        return None
+
+def handle_new_service(name, interface, service_info):
+    properties = normalize_dict_str(service_info.properties)
+    hostname = properties.get('liquid_hostname')
+    if not hostname:
+        return
+    is_local = (hostname == app.config['LIQUID_DOMAIN'])
+    if is_local:
+        address = get_ipv4_addr(interface)
```

Comment:
At this point, I think you can return `None`, because you're adding the local domain to dnsmasq [on line 112](https://github.com/liquidinvestigations/discover/pull/2/files#diff-f317a13e85ea3d1aff8e45a39d20aaffR112) anyway.